### PR TITLE
MF-282 - Rename clients to things in Dashflux

### LINKED ISSUE #282: Rename clients to things in Dashflux
Pull request #281 renamed `clients` to `things`. Besides renaming the entity and the service, all of the API references are updated as well. Make sure that the Dashflux API clients and its structure mimic these changes.

### PR DESCRIPTION
**Pull request by dborovcanin**
This pull request resolves #282.

### CODE REVIEW

File: dashflux/src/app/app.module.ts [hunk 1]
```diff
@@ -16,19 +16,19 @@ import { LoginComponent } from './components/auth/login/login.component';
...
 import { ConfirmationDialogComponent } from './components/shared/confirmation-dialog/confirmation-dialog.component';
 import { MaterialModule } from './core/material/material.module';
 import { AuthenticationService } from './core/services/auth/authentication.service';
 import { TokenStorage } from './core/services/auth/token-storage.service';
 import { ChannelsService } from './core/services/channels/channels.service';
-import { ClientsService } from './core/services/clients/clients.service';
+import { ThingsService } from './core/services/clients/clients.service';
 import { MockAuthService } from './core/services/mock-auth.service';
 import { MockChannelsService } from './core/services/mock-channels.service';
 import { MockClientsService } from './core/services/mock-clients.service';
```
Comment (nmarcetic) [inline dashflux/src/app/app.module.ts]: Rename.

File: dashflux/src/app/components/channels/add-channel-dialog/add-channel-dialog.component.html [hunk 2]
```diff
@@ -11,12 +11,12 @@ <h1 *ngIf="editMode" mat-dialog-title>Edit channel</h1>
       </mat-form-field>
       <mat-form-field>
           <mat-select placeholder="Connected clients" formControlName="connected" [compareWith]="compareFunction" multiple>
-            <mat-option *ngFor="let client of clientsStore.clients" [value]="client">{{client.name}}</mat-option>
+            <mat-option *ngFor="let client of clientsStore.things" [value]="client">{{client.name}}</mat-option>
```
Comment (nmarcetic) [inline dashflux/src/app/components/channels/add-channel-dialog/add-channel-dialog.component.html]: Rename.

File: dashflux/src/app/components/channels/add-channel-dialog/add-channel-dialog.component.spec.ts [hunk 3]
```diff
@@ -9,9 +9,9 @@ import { MaterialModule } from '../../../core/material/material.module';
 import { AuthenticationService } from '../../../core/services/auth/authentication.service';
 import { TokenStorage } from '../../../core/services/auth/token-storage.service';
 import { ChannelsService } from '../../../core/services/channels/channels.service';
-import { ClientsService } from '../../../core/services/clients/clients.service';
+import { ThingsService } from '../../../core/services/clients/clients.service';
```
Comment (nmarcetic) [inline dashflux/src/app/components/channels/add-channel-dialog/add-channel-dialog.component.spec.ts]: Rename.

File: dashflux/src/app/components/channels/add-channel-dialog/add-channel-dialog.component.ts [hunk 4]
```diff
@@ -20,7 +20,7 @@ export class AddChannelDialogComponent implements OnInit {
     private fb: FormBuilder,
     private dialogRef: MatDialogRef<AddChannelDialogComponent>,
     @Inject(MAT_DIALOG_DATA) public data: Channel,
-    public clientsStore: ClientsStore,
+    public clientsStore: ThingsStore,
```
Comment (nmarcetic) [inline dashflux/src/app/components/channels/add-channel-dialog/add-channel-dialog.component.ts]: Rename.

File: dashflux/src/app/components/channels/channels.component.spec.ts [hunk 5]
```diff
@@ -9,9 +9,9 @@ import { MaterialModule } from '../../core/material/material.module';
 import { AuthenticationService } from '../../core/services/auth/authentication.service';
 import { TokenStorage } from '../../core/services/auth/token-storage.service';
 import { ChannelsService } from '../../core/services/channels/channels.service';
-import { ClientsService } from '../../core/services/clients/clients.service';
+import { ThingsService } from '../../core/services/clients/clients.service';
```
Comment (nmarcetic) [inline dashflux/src/app/components/channels/channels.component.spec.ts]: Rename.

File: dashflux/src/app/components/channels/channels.component.ts [hunk 6]
```diff
@@ -19,13 +19,13 @@ export class ChannelsComponent implements OnInit {
 
   constructor(
     private dialog: MatDialog,
-    public clientsStore: ClientsStore,
+    public clientsStore: ThingsStore,
```
Comment (nmarcetic) [inline dashflux/src/app/components/channels/channels.component.ts:22]: Rename.

File: dashflux/src/app/components/channels/channels.component.ts [hunk 7]
```diff
@@ -19,13 +19,13 @@ export class ChannelsComponent implements OnInit {
...
     public channelsStore: ChannelsStore,
   ) { }
 
   ngOnInit() {
     this.channelsStore.getChannels();
-    this.clientsStore.getClients();
+    this.clientsStore.getThings();
```
Comment (nmarcetic) [inline dashflux/src/app/components/channels/channels.component.ts:28]: Rename.

File: dashflux/src/app/components/things/add-thing-dialog/add-thing-dialog.component.spec.ts [hunk 8]
```diff
@@ -9,16 +9,16 @@ import { MaterialModule } from '../../../core/material/material.module';
 import { AuthenticationService } from '../../../core/services/auth/authentication.service';
 import { TokenStorage } from '../../../core/services/auth/token-storage.service';
 import { ChannelsService } from '../../../core/services/channels/channels.service';
-import { ClientsService } from '../../../core/services/clients/clients.service';
-import { AddClientDialogComponent } from './add-client-dialog.component';
+import { ThingsService } from '../../../core/services/clients/clients.service';
```
Comment (nmarcetic) [inline dashflux/src/app/components/things/add-thing-dialog/add-thing-dialog.component.spec.ts]: Rename.

File: dashflux/src/app/components/things/things.component.spec.ts [hunk 9]
```diff
@@ -9,19 +9,19 @@ import { MaterialModule } from '../../core/material/material.module';
 import { AuthenticationService } from '../../core/services/auth/authentication.service';
 import { TokenStorage } from '../../core/services/auth/token-storage.service';
 import { ChannelsService } from '../../core/services/channels/channels.service';
-import { ClientsService } from '../../core/services/clients/clients.service';
-import { ClientsComponent } from './clients.component';
+import { ThingsService } from '../../core/services/clients/clients.service';
```
Comment (nmarcetic) [inline dashflux/src/app/components/things/things.component.spec.ts]: Rename.

File: dashflux/src/app/components/things/things.component.ts [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+import { ThingsStore } from '../../core/store/things.store';
+import { ChannelsStore } from '../../core/store/channels.store';
+
+@Component({
+  selector: 'app-things',
+  templateUrl: './things.component.html',
+  styleUrls: ['./things.component.scss']
+})
+export class ThingsComponent implements OnInit {
+  clients: Observable<Thing[]>;
```
Comment (nmarcetic) [inline dashflux/src/app/components/things/things.component.ts]: Rename.

File: dashflux/src/app/core/services/channels/channels.service.ts [hunk 11]
```diff
@@ -60,44 +60,44 @@ export class ChannelsService {
...
       if (clientsToDelete.length) {
-        return forkJoin(this.createClientDisconnectRequests(channel.id, clientsToDelete));
+        return forkJoin(this.createThingsDisconnectRequests(channel.id, clientsToDelete));
       } else {
         return Observable.of([]);
       }
     });
   }
 
   getClientsToDelete(channelFormData: Channel, channel: Channel) {
```
Comment (nmarcetic) [inline dashflux/src/app/core/services/channels/channels.service.ts]: Rename this and all above.

File: dashflux/src/app/core/services/mock-clients.service.ts [hunk 12]
```diff
@@ -1,9 +1,9 @@
 import { Injectable } from '@angular/core';
 import { Observable } from 'rxjs/Observable';
 
-import { Client } from '../store/models';
+import { Thing } from '../store/models';
 
-let MOCK_CLIENTS = [
+const MOCK_CLIENTS = [
```
Comment (nmarcetic) [inline dashflux/src/app/core/services/mock-clients.service.ts:6]: Rename.

File: dashflux/src/app/core/services/mock-clients.service.ts [hunk 13]
```diff
@@ -12,8 +12,8 @@ export class MockClientsService {
       return Observable.of(MOCK_CLIENTS).delay(1000);
```
Comment (nmarcetic) [inline dashflux/src/app/core/services/mock-clients.service.ts:12]: Rename.

File: dashflux/src/app/core/services/mock-clients.service.ts [hunk 14]
```diff
@@ -12,8 +12,8 @@ export class MockClientsService {
...
   }
 
-  addClient(client: Client) {
-    MOCK_CLIENTS.push(client);
+  addClient(thing: Thing) {
+    MOCK_CLIENTS.push(thing);
```
Comment (nmarcetic) [inline dashflux/src/app/core/services/mock-clients.service.ts]: Rename.

File: dashflux/src/app/core/store/auth.store.spec.ts [hunk 15]
```diff
@@ -7,7 +7,7 @@ import { Observable } from 'rxjs/Observable';
 import { AuthenticationService } from '../services/auth/authentication.service';
 import { TokenStorage } from '../services/auth/token-storage.service';
 import { ChannelsService } from '../services/channels/channels.service';
-import { ClientsService } from '../services/clients/clients.service';
+import { ThingsService } from '../services/clients/clients.service';
```
Comment (nmarcetic) [inline dashflux/src/app/core/store/auth.store.spec.ts]: Rename.

File: dashflux/src/app/core/store/things.store.spec.ts [hunk 16]
```diff
@@ -0,0 +1,256 @@
+import { HttpClientModule } from '@angular/common/http';
+import { inject, TestBed } from '@angular/core/testing';
+import { RouterTestingModule } from '@angular/router/testing';
+import { toJS } from 'mobx';
+import { Observable } from 'rxjs/Observable';
+
+import { ThingsService } from '../services/clients/clients.service';
```
Comment (nmarcetic) [inline dashflux/src/app/core/store/things.store.spec.ts]: Rename.

File: dashflux/src/app/core/store/things.store.ts [hunk 17]
```diff
@@ -1,62 +1,62 @@
 import { Injectable } from '@angular/core';
 import { action, observable } from 'mobx';
 
-import { ClientsService } from '../services/clients/clients.service';
-import { Client } from './models';
+import { ThingsService } from '../services/clients/clients.service';
```
Comment (nmarcetic) [inline dashflux/src/app/core/store/things.store.ts]: Rename.

File: dashflux/src/app/core/store/ui.store.spec.ts [hunk 18]
```diff
@@ -6,7 +6,7 @@ import { RouterTestingModule } from '@angular/router/testing';
 import { AuthenticationService } from '../services/auth/authentication.service';
 import { TokenStorage } from '../services/auth/token-storage.service';
 import { ChannelsService } from '../services/channels/channels.service';
-import { ClientsService } from '../services/clients/clients.service';
+import { ThingsService } from '../services/clients/clients.service';
```
Comment (nmarcetic) [inline dashflux/src/app/core/store/ui.store.spec.ts]: Rename.

**Review by nmarcetic — CHANGES_REQUESTED** [review]

**Comment by dborovcanin** [comment]
This PR is moved to #305.

**Pull request closed without merging**